### DISCUSSION TITLE
ERROR: Can't install RMagick 4.2.3. Can't find the ImageMagick library or one of the dependent libraries.

### DISCUSSION THREAD

**Discussion by ashishjg21** [alshedivat/al-folio#501] [Q&A] [answered]
Newbie here sorry 🙏
**Describe the bug**
bundle install gives this message...please help

ERROR: Can't install RMagick 4.2.3.
Can't find the ImageMagick library or one of the dependent libraries.

**Expected behavior**
I expected a normal bundle install setup of gem files.

**Screenshots**
[https://imgur.com/XEU3FL0](url)

**System :**
 - macOS Monterey 12.0.1
- Ruby version ruby 3.0.2p107 (2021-07-07 revision 0db68f0233) [x86_64-darwin21]

current directory: /Users/apple/.gem/ruby/3.0.2/gems/rmagick-4.2.3/ext/RMagick
/Users/apple/.rubies/ruby-3.0.2/bin/ruby -I /Users/apple/.rubies/ruby-3.0.2/lib/ruby/3.0.0 -r ./siteconf20211212-4929-7kx2rv.rb extconf.rb
checking for brew... yes
checking for Ruby version >= 2.3.0... yes
checking for pkg-config... yes
Package MagickCore was not found in the pkg-config search path.
Perhaps you should add the directory containing `MagickCore.pc'
to the PKG_CONFIG_PATH environment variable
No package 'MagickCore' found

ERROR: Can't install RMagick 4.2.3.
Can't find the ImageMagick library or one of the dependent libraries.
Check the mkmf.log file for more detailed information.

*** extconf.rb failed ***
Could not create Makefile due to some reason, probably lack of necessary
libraries and/or headers.  Check the mkmf.log file for more details.  You may
need configuration options.

Provided configuration options:
	--with-opt-dir
	--with-opt-include
	--without-opt-include=${opt-dir}/include
	--with-opt-lib
	--without-opt-lib=${opt-dir}/lib
	--with-make-prog
	--without-make-prog
	--srcdir=.
	--curdir
	--ruby=/Users/apple/.rubies/ruby-3.0.2/bin/$(RUBY_BASE_NAME)

To see why this extension failed to compile, please check the mkmf.log which can be found here:

  /Users/apple/.gem/ruby/3.0.2/extensions/x86_64-darwin-21/3.0.0-static/rmagick-4.2.3/mkmf.log

extconf failed, exit code 1

Gem files will remain installed in /Users/apple/.gem/ruby/3.0.2/gems/rmagick-4.2.3 for inspection.
Results logged to /Users/apple/.gem/ruby/3.0.2/extensions/x86_64-darwin-21/3.0.0-static/rmagick-4.2.3/gem_make.out

An error occurred while installing rmagick (4.2.3), and Bundler cannot continue.
Make sure that `gem install rmagick -v '4.2.3' --source 'https://rubygems.org/'` succeeds before bundling.

In Gemfile:
  jekyll-responsive-image was resolved to 1.6.0, which depends on
    rmagick

**Comment by RafalKucharskiPK** [comment]
did you try automated deployment via github actions?

**Reply by ashishjg21** [reply]
You mean we can deploy directly without cloning to local or running it in local environment?

Can you please tell how to do?or link to some blog which shows the steps? (or video)

**Comment by rohandebsarkar** [comment, marked as answer]
Try installing ImageMagick from: https://imagemagick.org/script/download.php

Detailed steps for macOS X: https://imagemagick.org/script/download.php#macosx

**Reply by rohandebsarkar** [reply]
If you are using brew: `brew install imagemagick`

**Reply by capncapes** [reply]
This works beautifully. Thanks!
